app/controllers/waitlist_team_controller.rb

```diff
@@ -0,0 +1,103 @@
+class WaitlistTeamController < ApplicationController
```

Comment:
The name `WaitlistTeamController` doesn't make much sense to me, so does `WaitlistTeam` model. Perhaps this should be renamed as `WaitlistController` and `Waitlist` model?